pkg/component/worker/kubelet.go

```diff
@@ -131,6 +132,27 @@ func (k *Kubelet) Init(_ context.Context) error {
 	return nil
 }
 
+func getIPs(ctx context.Context, hostname string) (net.IP, net.IP, error) {
+	ipaddrs, err := net.DefaultResolver.LookupIPAddr(ctx, hostname)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ipv4, ipv6 net.IP
+	for _, addr := range ipaddrs {
+		if ipv4 == nil && addr.IP.To4() != nil {
+			ipv4 = addr.IP
```

Comment:
```suggestion
			ipv4 = addr.IP.To4()
```

Reply:
What would this solve?

Reply:
It would ensure that ipv4 is actually in its four byte form, so that `len(ipv4) == IPv4len`. Otherwise, it could be in its 16 byte form, i.e. `len(ipv4) == IPv6len`. This is a nit, but it feels a bit saner to have a v4 IP returned as 4 bytes.

Reply:
I think we only care what the address family is. How the buffers are allocated under the hood is completely irrelevant.